src/shogun/distributions/KernelDensity.cpp

```diff
@@ -91,12 +91,14 @@ bool KernelDensity::train(std::shared_ptr<Features> data)
 	return true;
 }
 
-SGVector<float64_t> KernelDensity::get_log_density(const std::shared_ptr<DenseFeatures<float64_t>>& test, int32_t leaf_size)
+SGVector<float64_t> KernelDensity::get_log_density(const std::shared_ptr<Features>& test, int32_t leaf_size)
```

Comment:
I don't really see the point of moving this check to runtime?

Reply:
@karlnapf unified/simplified API ;)

Reply:
features() returns Features... you wont be able to call this from SWIG API, as you dont have RealFeatures anymore...

Reply:
sure thing. I didnt realise this is exposed to swig.

Reply:
well KDE notebook uses it :S

Reply:
i guess there could be a way to refactor this even more so the function is not exposed at all, but on the end somehow features should get to this object, i.e. it'll have to be general Features as well...

Reply:
yeah, the public interface needs the base class for sure. I think this is good in the meantime and at some point we could/should refactor the distribution api....
Also with this, we can add a meta example for distributions!